### ISSUE TITLE
Accessibility improvements for default project template

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/aspnetcore#32390]
I've been trying to use the Blazor default project template through a screen reader and it's not a good experience. There are probably dozens of ways it could be improved but the most striking things were:

 * The homepage is really annoying because of the `<SurveyPrompt>` having a `role="alert"`. This causes it to take precedence over everything else and gets read out first as if that's the only important content.
 * Navigation is extremely hard to make any sense of, because there are no announcements when navigation happens. When you click on a link, literally nothing happens as far as the screen reader is concerned. New page titles should be announced (like they would be if it was a full-page load)
 * The "counter" sample behaves poorly. When you click the button, the screen reader says nothing. We should consider adding `aria-live` to the "current count" display so it gets read out on each update.

Partly this should be resolved by improvements in the template, but partly we should look at ensuring the right infrastructure exists for doing this well. For example, we might want the router to know about how to announce navigation events. We might also want to consider adding some focus-management features (e.g., ability to constrain keyboard focus within an element for things like dialogs, and a simpler way to auto-focus particular elements when they first get rendered, maybe even consider something like `@bind-focus="shouldBeFocused"`).

**Comment by SteveSandersonMS** [comment]
Done in https://github.com/dotnet/aspnetcore/pull/33474 and https://github.com/dotnet/aspnetcore/pull/33470
